metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -92,6 +89,7 @@ final class BloopServers(
   // Returns a random free port.
   private def randomPort(host: String): Int = {
     val s = new ServerSocket()
+    s.setReuseAddress(true)
```

Comment:
Interesting, what is this useful for?

Reply:
The OS will allow you to reopen a closed socket within a specific timeframe if you set that flag. Depending on the OS it's either when you first open it, or when you reopen it. Pretty good explanation here - https://stackoverflow.com/questions/14388706/socket-options-so-reuseaddr-and-so-reuseport-how-do-they-differ-do-they-mean-t

Reply:
And sure, I can do a PR on the launcher, no problem.

Reply:
My previous attempts to move to the Bloop launcher have so far failed to pass CI. I believe if we use the launcher as a library instead of shelling out to it then it will work better, and we could clean up a lot of hairy code in Metals.